xtask/src/checks.rs

```diff
@@ -0,0 +1,38 @@
+use crate::{args::FlagExt, Result};
+use clap::Parser;
+use xshell::{cmd, Shell};
+
+#[derive(Debug, Parser)]
+pub struct Checks {
+    #[clap(long)]
+    enforce_warnings: Option<Option<bool>>,
+
+    #[clap(long, default_value = "nightly")]
+    rustfmt_toolchain: String,
+}
+
+impl Checks {
+    pub fn run(&self, sh: &Shell) -> Result {
+        crate::build::Build {
+            release: Some(Some(false)),
+        }
+        .run(sh)?;
+
+        {
+            let toolchain = format!("+{}", self.rustfmt_toolchain);
+            cmd!(sh, "cargo {toolchain} fmt --all -- --check").run()?;
+        }
```

Comment:
Why is the scope necessary?

Reply:
i just didn't want to pollute the whole function body with variables that aren't needed